src/popup.tsx

```diff
@@ -1,198 +1,58 @@
-            <Route path="/quick-settings/tokens/:id">
-              {(params: { id: string }) => <TokenSettings id={params?.id} />}
-            </Route>
-          </Switch>
-          <Route path="/quick-settings/contacts" component={Contacts} />
-          <Switch>
-            <Route path="/quick-settings/contacts/new" component={NewContact} />
-            <Route path="/quick-settings/contacts/:address">
-              {(params: { address: string }) => (
-                <ContactSettings address={params?.address} />
-              )}
-            </Route>
-          </Switch>
-          <Route
-            path="/quick-settings/notifications"
-            component={NotificationSettings}
-          />
-          <Route path="/subscriptions/:id">
-            {(params: { id: string }) => (
-              <SubscriptionDetails id={params?.id} />
-            )}
-          </Route>
-          <Route path="/subscriptions/:id/manage">
-            {(params: { id: string }) => (
-              <SubscriptionManagement id={params?.id} />
-            )}
-          </Route>
-          <Route path="/subscriptions/:id/payment">
-            {(params: { id: string }) => (
-              <SubscriptionPayment id={params?.id} />
-            )}
-          </Route>
-          <Route path="/transactions" component={Transactions} />
-          <Route path="/notifications" component={Notifications} />
-          <Route path="/notification/:id">
-            {(params: { id: string }) => (
-              <MessageNotification id={params?.id} />
-            )}
-          </Route>
-          <Route path="/tokens" component={Tokens} />
-          <Route path="/token/:id">
-            {(params: { id: string }) => <Asset id={params?.id} />}
-          </Route>
-          <Route path="/collectibles" component={Collectibles} />
-          <Route path="/collectible/:id">
-            {(params: { id: string }) => <Collectible id={params?.id} />}
-          </Route>
-          <Route path="/transaction/:id/:gateway?">
-            {(params: { id: string; gateway?: string }) => (
-              <Transaction id={params?.id} gw={params?.gateway} />
-            )}
-          </Route>
-          <Route path="/send/confirm/:token/:qty/:recipient/:message?">
-            {(params: { token: string; qty: string }) => (
-              <Confirm
-                tokenID={params?.token}
-                qty={Number(params?.qty || "0")}
-              />
-            )}
-          </Route>
-          <Route path="/send/recipient/:token/:qty/:message?">
-            {(params: { token: string; qty: string; message?: string }) => (
-              <Recipient
-                tokenID={params?.token}
-                qty={params?.qty || "0"}
-                message={params?.message || ""}
-              />
-            )}
-          </Route>
-          <NavigationBar />
-        </HistoryProvider>
-      </Router>
+      <>
+        <Routes routes={POPUP_ROUTES} />
```

Comment:
Routes have been defined in `POPUP_ROUTES` and are wrapped automatically in a `Page` component by the `Routes` component.

Additionally, the `Head` component could be extracted here (just like `NavigationBar`) so that it is not affected by page transitions, but that would also require adding a `headProps` property to the `RouterConfig` interface (for static personalization) and also creating a context and/or portal to allow for personalization coming directly from the view/page's own life cycle (for dynamic personalization).